src/sync.ts

```diff
@@ -330,16 +331,26 @@ export class Sync {
             if (fileToUpload.fileName === "cloudSettings") {
               return false;
             }
-            if (!gistObj.data.files[fileToUpload.fileName]) {
+            if (!gistObj.data.files[fileToUpload.fileName] &&
```

Comment:
Why are these checks are changed ? 
If you are looking to fix the `snippets` sync bug. Its already fixed in the branch. You can verify by pulling.

Reply:
This is a different problem with snippets.

This code is checking to see if there are local files that are not in the gist yet or if any of the local files have content that is different from what is in the gist.

The short story is that when there are snippets or any files that are in a subdir, this code will always return true because the `fileToUpload.fileName` is not the same as the gist's filename when the file is in a subdir.  The code then thinks that the local file is not in the gist and returns true.  When this function returns true, there will always be an upload even if no content changed.  This results in an "empty" gist revision with only the upload time changed.

To see this in action, add:
```typescript
console.log(fileToUpload.gistName);
```
somewhere inside of the function, and you will see:
```
extensions.json
keybindings.json
settings.json
snippets|c.json
snippets|cpp.json
```
if you have `c.json` and `cpp.json` snippets.  Note the "|" in use as a directory separator.

When I first pushed this, I didn't know about the `gistName` member.  A _much_ better fix for this would be to use `fileToUpload.gistName` to index `gistObj.data.files[]` instead of trying to construct the gist name from the dir and the "|" like I did here.

I will update this PR with this better fix.

BTW, I had to revert 7ae30632d7b6719c04d594a76d0e66977042dad6 in order to upload settings.  I think you can see why after this discussion.

Reply:
Alright. Thanks for the explanation. 
I will test it.